app/models/page_template.rb

```diff
@@ -5,12 +5,14 @@ class PageTemplate < ApplicationRecord
 
   has_many :pages, foreign_key: 'template_id',
                    inverse_of: 'template',
-                   dependent: :delete_all
+                   dependent: :nullify
   has_many :elements, class_name: 'PageTemplateElement',
                       foreign_key: 'template_id',
                       inverse_of: 'template',
                       dependent: :delete_all
 
+  accepts_nested_attributes_for :elements
```

Comment:
👍 for `accepts_nested_attributes_for` usage

Reply:
I've just written up [the blog post for this PR](http://shinycms.org/blog/2019/10/magic) and a year later, I'm still pleased with myself for actively seeking out and finding (some sort of magic in the region of) `accepts_nested_attributes_for` (and getting it working), exactly when I needed it, all on my own :stuck_out_tongue: This was the first time I felt like a good native Rails developer rather than a barely competent non-idiomatic one :slightly_smiling_face: 

This was when the project proved itself as a learning tool for me (a lot earlier than I had any right to expect or even hope, really!) - also, ironically, this is when I got interested enough and confident enough that it stopped being just a personal educational project and turned into a thing I wanted to see through to a release. Which of course has made it all the more educational. :smiley: